src/sentry/digests/utilities.py

```diff
@@ -23,3 +24,134 @@ def get_digest_metadata(digest):
+            events.append(group_records[0].value.event)
+    return set(events)
+
+
+def build_custom_digest(original_digest, events):
+    """
+    build_custom_digest(original_digest: Digest, user_id: Int, events_by_users: Map[User_Id:Set(Events)]) -> Digest
+    """
+    user_digest = OrderedDict()
+    for rule, rule_groups in six.iteritems(original_digest):
+        user_rule_groups = OrderedDict()
+        for group, group_records in six.iteritems(rule_groups):
+            user_group_records = [
+                record for record in group_records
+                if record.value.event in events
+            ]
+            if user_group_records:
+                user_rule_groups[group] = sort_records(user_group_records)
+        if user_rule_groups:
+            user_digest[rule] = user_rule_groups
+    return user_digest
+
+
+def build_events_by_actor(project_id, events):
+    """
+    build_events_by_actor(project_id: Int, events: Set(Events)) -> Map[Actor:Set(Events)]
+    """
+    events_by_actor = {}
+    for event in events:
+        # TODO(LB): I Know this is inefficent.
+        # ProjectOwnership.get_owners is O(n) queries and I'm doing that O(len(events)) times
+        # I will create a follow-up PR to address this method's efficency problem
+        # Just wanted to make as few changes as possible for now.
+        actors, __ = ProjectOwnership.get_owners(project_id, event.data)
+        if actors == ProjectOwnership.Everyone:
+            actors = [actors]
+        for actor in actors:
+            if actor in events_by_actor:
+                events_by_actor[actor].add(event)
+            else:
+                events_by_actor[actor] = set([event])
+    return events_by_actor
+
+
+def convert_actors_to_user_set(events_by_actor, user_ids):
+    """
+    convert_actors_to_user_set(events_by_actor: Map[Actor:Set(Events)], user_ids: List(Int)) -> Map[User_Id:Set(Events)]
+    """
+    def add_user_id(user_id, events, user_by_events):
+        if user_id in user_by_events:
+            user_by_events[user_id].update(set(events))
+        else:
+            user_by_events[user_id] = set(events)
+
+    user_by_events = {}
+    if ProjectOwnership.Everyone in events_by_actor:
+        for user_id in user_ids:
+            user_by_events[user_id] = set(events_by_actor[ProjectOwnership.Everyone])
+        del events_by_actor[ProjectOwnership.Everyone]
+
+    team_actors = [actor for actor in six.iterkeys(events_by_actor) if actor.type == Team]
+    teams_to_user_ids = team_actors_to_user_ids(team_actors, user_ids)
+    for actor, events in six.iteritems(events_by_actor):
+        if actor.type == Team:
+            try:
+                user_ids = teams_to_user_ids[actor.id]
+            except KeyError:
+                # TODO(LB): Not certain what to do if a team has no active members
+                # Created a Test to reflect my assumptions here.
+                pass
+            else:
```

Comment:
It's better for future-proofing to construct this similar to a switch statement, checking here explicitly for `actor.type == User` and include an `else` block (equivalent to a `default` case) that raises an error if an actor is provided that isn't one of the expected `Team` or `User` types.

Reply:
This should probably do something with unknown actor types (the `else` block mentioned in the above comment) — log that it received a value it didn't understand and ignored it, raise an error, etc — instead of silently discarding them.